cli/keys.go

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+package cli
+
+import (
+	"time"
+
+	"github.com/spf13/cobra"
+)
+
+var cmdAPIKeys = []cobra.Command{
+	{
+		Use:   "issue",
+		Short: "issue <duration> <user_auth_token>",
+		Long:  `Issues a new Key`,
+		Run: func(cmd *cobra.Command, args []string) {
+			if len(args) != 2 {
+				logUsage(cmd.Short)
+				return
+			}
+
+			d, err := time.ParseDuration(args[0])
+			if err != nil {
+				logError(err)
+				return
+			}
+
+			resp, err := sdk.Issue(args[1], d)
+			if err != nil {
+				logError(err)
+				return
+			}
+
+			logJSON(resp)
+		},
+	},
+	{
+		Use:   "revoke",
+		Short: "revoke <key_id> <user_auth_token>",
+		Long:  `Removes API key from database`,
+		Run: func(cmd *cobra.Command, args []string) {
+			if len(args) != 2 {
+				logUsage(cmd.Short)
+				return
+			}
+
+			if err := sdk.Revoke(args[0], args[1]); err != nil {
+				logError(err)
+				return
+			}
+
+			logOK()
+		},
+	},
+	{
+		Use:   "retrieve",
+		Short: "retrieve <key_id> <user_auth_token>",
+		Long:  `Retrieves API key with given id`,
+		Run: func(cmd *cobra.Command, args []string) {
+			if len(args) != 2 {
+				logUsage(cmd.Short)
+				return
+			}
+
+			rk, err := sdk.RetrieveKey(args[0], args[1])
+			if err != nil {
+				logError(err)
+				return
+			}
+
+			logJSON(rk)
+		},
+	},
+}
+
+// NewKeysCmd returns keys command.
+func NewKeysCmd() *cobra.Command {
+	cmd := cobra.Command{
+		Use:   "keys",
+		Short: "Keys management",
+		Long:  `Keys management: issue, revoke, or retrieve API key.`,
+		Run: func(cmd *cobra.Command, args []string) {
+			logUsage("keys [issue | revoke | retrieve]")
```

Comment:
In other commands this is defined as `cmd.Short`

Reply:
In functions NewGroupsCmd, NewThingsCmd, NewChannlesCmd it is not defined as cmd.Short..
For example : https://github.com/mainflux/mainflux/blob/42dd813521e11b5ec2cac07509b07042ef35d5b7/cli/groups.go#L212

Reply:
Right, it's consistent. But in fact this wasn't properly done in a first place. Firstly, we should use `cmd.Short` since this can be seen with `--help` flag while now it will be only displayed if you type a wrong command. Secondly, we should replace `cmd.Short` by `cmd.Use` following this Cobra dicctionnary: 
```
	// Use is the one-line usage message.
	// Recommended syntax is as follow:
	//   [ ] identifies an optional argument. Arguments that are not enclosed in brackets are required.
	//   ... indicates that you can specify multiple values for the previous argument.
	//   |   indicates mutually exclusive information. You can use the argument to the left of the separator or the
	//       argument to the right of the separator. You cannot use both arguments in a single use of the command.
	//   { } delimits a set of mutually exclusive arguments when one of the arguments is required. If the arguments are
	//       optional, they are enclosed in brackets ([ ]).
	// Example: add [-F file | -D dir]... [-f format] profile
	Use string
```

ILike that when you we use the `--help` flag the generated doc would be complete.

If you want I can do it in a separate PR.  Or maybe you can take this one?

Reply:
I would say it's better to do it in a separate PR so that we keep this one focused on API keys.

Reply:
Lets do it in separate PR and we should incloude this issue in that PR : https://github.com/mainflux/mainflux/issues/1551.

Reply:
@dusanb94 @fbugarski Ok, I can take both PRs.